pkg/hubble/container/ring_reader.go

```diff
@@ -77,26 +85,44 @@ func (r *RingReader) Next() (*v1.Event, error) {
 func (r *RingReader) NextFollow(ctx context.Context) *v1.Event {
 	// if the context changed between invocations, we also have to restart
 	// readFrom, as the old readFrom instance will be using the old context.
-	if r.c == nil || r.ctx != ctx {
-		r.c = r.ring.readFrom(ctx, r.idx)
+	if r.ctx != ctx {
+		if r.followChan == nil {
+			r.followChan = make(chan *v1.Event, r.followChanLen)
```

Comment:
I find it a bit unergonomic that in order to stop the ring reader properly, you always have to both cancel the context and close the ring reader. I think if you switch the context between calls to NextFollow, the old channel can actually never be closed. Is that an issue?

I have not thought about it too hard, but it be possible to close `r.followChan` once the context is cancelled like in the previous version. We still could have a `Close()` method used for testing.

Reply:
That's a good feedback. I updated the code to make `Close()` optional as it's mostly useful for testing only.

Reply:
Awesome, one minor follow-up below